src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/DynamicMethod.cs

```diff
@@ -431,38 +465,119 @@ internal RuntimeMethodHandle GetMethodDescriptor()
             _ = GetMethodDescriptor();
             // ignore obj since it's a static method
 
-            // create a signature object
-            Signature sig = new Signature(
-                this.m_methodHandle!, m_parameterTypes, m_returnType, CallingConvention);
-
-
             // verify arguments
-            int formalCount = sig.Arguments.Length;
-            int actualCount = (parameters != null) ? parameters.Length : 0;
-            if (formalCount != actualCount)
+            int argCount = (parameters != null) ? parameters.Length : 0;
+            if (Signature.Arguments.Length != argCount)
                 throw new TargetParameterCountException(SR.Arg_ParmCnt);
 
-            // if we are here we passed all the previous checks. Time to look at the arguments
-            bool wrapExceptions = (invokeAttr & BindingFlags.DoNotWrapExceptions) == 0;
+            object? retValue;
 
-            StackAllocedArguments stackArgs = default;
-            Span<object?> arguments = default;
-            if (actualCount != 0)
+            unsafe
             {
-                arguments = CheckArguments(ref stackArgs, parameters, binder, invokeAttr, culture, sig.Arguments);
-            }
+                if (argCount == 0)
+                {
+                    retValue = Invoker.InvokeUnsafe(obj, args: default, invokeAttr);
+                }
+                else
+                {
+                    Debug.Assert(parameters != null);
+                    Span<object?> copyOfParameters;
+                    Span<bool> shouldCopyBackParameters;
+
+                    if (argCount <= MaxStackAllocArgCount)
+                    {
+                        StackAllocedArguments argStorage = default;
+                        copyOfParameters = new Span<object?>(ref argStorage._arg0, argCount);
+                        shouldCopyBackParameters = new Span<bool>(ref argStorage._copyBack0, argCount);
+
+                        StackAllocatedByRefs byrefStorage = default;
+
+                        CheckArguments(
+                            copyOfParameters,
+                            (IntPtr*)&byrefStorage,
+                            shouldCopyBackParameters,
+                            parameters,
+                            Signature.Arguments,
+                            binder,
+                            culture,
+                            invokeAttr);
+
+                        retValue = Invoker.InvokeUnsafe(obj, (IntPtr*)(void**)&byrefStorage, invokeAttr);
+                    }
+                    else
+                    {
+                        object[] objHolder = new object[argCount];
```

Comment:
It would likely help performance if this path for large number of arguments was factored into in a separate method. I expect that the stackallocs and try/finally here are negatively affecting the fast path.

Reply:
I moved it to a local function; I'll also do another perf pass once the Emit support is there.